test_function_calling.py

```diff
@@ -0,0 +1,106 @@
+import requests
+import json
+
+# Define the endpoint and headers
+url = "<endpoint-url>"
+headers = {
+    'Authorization': 'Bearer <token>',
+    # 'azureml-model-deployment': 'llama31-8b--deployment-aml-1', (required only when deployed on AzureML)
+    'accept': 'application/json',
+    'Content-Type': 'application/json'
+}
+
+# Define the initial messages
+messages = [
+    {"role": "user", "content": "Hello! How are you?"},
+    {"role": "assistant", "content": "Hi! I am quite well, how can I help you today?"},
+    {"role": "user", "content": "Can you get me weather for San Francisco, CA?"}
+]
+
+# Define function
+weather_tool = {
+    "type": "function",
+    "function": {
+        "name": "get_current_weather",
+        "description": "Get the current weather",
+        "parameters": {
+            "type": "object",
+            "properties": {
+                "location": {
+                    "type": "string",
+                    "description": "The city and state, e.g. San Francisco, CA"
+                },
+                "format": {
+                    "type": "string",
+                    "enum": ["celsius", "fahrenheit"],
+                    "description": "The temperature unit to use. Infer this from the user's location."
+                }
+            },
+            "required": ["location", "format"]
+        }
+    }
+}
+
+def get_weather(location, format):
+    # Dummy weather data for predefined cities
+    dummy_weather_data = {
+        "San Francisco, CA": {"temperature": 68, "unit": "fahrenheit"},
+        "Paris, France": {"temperature": 20, "unit": "celsius"}
+    }
+    
+    # Return the weather data if the location is known
+    if location in dummy_weather_data:
+        return dummy_weather_data[location]
+    else:
+        return {"error": "Weather data not available for the specified location"}
+
+def make_request(messages, tools=None, tool_choice=None):
+    data = {
+        "model": "meta/llama-3_1-8b-instruct",
```

Comment:
Can we parameterize the model name here so that this script can be used for other models or new models that come out with llama3.1-8b being the default?

Reply:
Modified to make it a callable python script.